engine/runtime-integration-tests/src/test/java/org/enso/compiler/test/passes/resolve/FullyQualifiedNamesTest.java

```diff
@@ -0,0 +1,135 @@
+package org.enso.compiler.test.passes.resolve;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Set;
+import java.util.function.Consumer;
+import org.enso.common.RuntimeOptions;
+import org.enso.compiler.MetadataInteropHelpers;
+import org.enso.compiler.core.IR;
+import org.enso.compiler.core.ir.Location;
+import org.enso.compiler.core.ir.Module;
+import org.enso.compiler.pass.resolve.FullyQualifiedNames;
+import org.enso.compiler.pass.resolve.FullyQualifiedNames$;
+import org.enso.pkg.QualifiedName;
+import org.enso.polyglot.PolyglotContext;
+import org.enso.test.utils.ContextUtils;
+import org.enso.test.utils.ProjectUtils;
+import org.enso.test.utils.SourceModule;
+import org.graalvm.polyglot.Context;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public final class FullyQualifiedNamesTest {
+
+  @ClassRule public static final TemporaryFolder TMP_DIR = new TemporaryFolder();
+
+  private Path projDir;
+
+  @Before
+  public void before() throws IOException {
+    var tmpDir = TMP_DIR.newFolder();
+    projDir = tmpDir.toPath().resolve("Proj");
+    var wasCreated = projDir.toFile().mkdir();
+    assert wasCreated;
+  }
+
+  @After
+  public void after() {}
+
+  @Test
+  public void libraryNameIsResolved_InExpressionBlock() throws IOException {
+    String mainSrc = """
+        main =
+            local.Proj.My_Module.My_Type
```

Comment:
So far, `FullyQualifiedNames` compiler pass is able to only append metadata to `local.Proj`, but not to the whole expression. However, during compilation, we could resolve the whole expression here.